### ISSUE TITLE
Black Screen on OLED65C3PUA

### ISSUE THREAD

**Issue by sethly87** [Cathwyler/MagicRemoteService#12]
Hello, I've gone through the instructions and have successfully set up both TV and PC with the service. My PC is plugged in through a Denon AVR to the TV but only a black screen shows when the TV app is active.

I can RDP into my PC I can see that the remote is registering mouse movement and button presses corresponding to the TV cursor which is visible on the TV display. For what it's worth, I do not see a small white box like other issues have mentioned.

Model: OLED65C3PUA
WebOS Version: 8.2.0-710

**Comment by Cathwyler** [comment]
Hi,

The TV app is showing the input you selected in the TV tab on the PC app. This app is not a video streaming solution.
If you change the selected input, for change to take effect you need to reinstall the TV app.

I don't know exactly how you wired your PC to your TV, but i supose you need to select the HDMI input of your Denon AVR. With your Denon AVR turned on and showing the PC input.

Regards

**Comment by sethly87** [comment]
The inout on the TV app is correct; it pulls the proper display name and everything. When I "Install on TV" the magicremoteservice app starts on the TV and this is when the remote functions as a mouse on the computer, but the display is completely black on the TV with only the remote cursor visible. When I switch to the HDMI input via the TV input method, then I see the display but the remote no longer works. 

Something I noticed after making all the connections and enabling the key server option in the dev mode app, it now no longer stays on. Should key server remain on?

I reread the instructions several times and believe I followed all of it.

**Comment by Cathwyler** [comment]
You need to select the correct HDMI number wired (the one on the TV) before install on TV.
![Sans titre](https://github.com/Cathwyler/MagicRemoteService/assets/4673422/bdaf7174-c569-49e0-840b-5bd898df2986)
You can install multiple MagicRemoteService app, one for each input. 
Maybe you have two or more MagicRemoteService installed ? If yes you can delete them all and retry.
You no longer need to use HDMi app but MagicRemoteService instead. MagicRemoteService act like an HDMI input.

**Comment by sethly87** [comment]
Thank you for the reply! If that is the case, is an update/workaround in the scope of this project for the future?

Regardless, thank you for the work you have done on this service; it was the exact feature I wanted when setting up my home theater.

**Comment by Cathwyler** [comment]
Hi,

I just added a new overlay option over native hdmi app. That can maybe fix your issue in this pre-release [1.2.4.0](https://github.com/Cathwyler/MagicRemoteService/releases/tag/1.2.4.0)

Not that this option is fully experimental with plenty of unofficial webOS option. It works on webOS 6.0.
Let me know if it works fine.

Regards

**Comment by sethly87** [comment]
Thank you, I will test this out today! And to clarify, as I'm not an expert in how the service vs the setup UI works, is this the correct process for update and use?

1. Stop MagicRemoteService service in Task Manager
2. Close program in the icons list if running
3. Replace all files in the installation directory
4. Start the service
5. Run MagicRemoteService.exe as administrator from the bin menu (unsure of this step as I've also seen the exe in another folder)
6. Confirm TV and PC details, install app to TV

What is the most effective way to check that the correct version is running?

**Comment by Cathwyler** [comment]
Stop the MagicRemoteService from Windows services list (service.msc)
Check task manager and kill all others remaning instances
Replace all files in the installation directory
Run MagicRemoteService.exe in the installation directory
Install on TV and save PC details, (This step will relaunch windows service)

I will add a the version number in the settings form in a futur release to be sure the good one i running.

**Comment by sethly87** [comment]
A couple things:

1. Should I be rebuilding through build.bat at all during reinstallation? I tried and got the following build errors `Build FAILED.

"C:\MagicRemoteService\MagicRemoteService.sln" (default target) (1) ->
"C:\MagicRemoteService\MagicRemoteService\MagicRemoteService.csproj" (default target) (2) ->
(CoreResGen target) ->
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.tar
gets(3162,5): error MSB3821: Couldn't process file BindCreator.resx due to its being in the Internet or Restricted zone
 or having the mark of the web on the file. Remove the mark of the web if you want to process these files. [C:\MagicRem
oteService\MagicRemoteService\MagicRemoteService.csproj]
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.tar
gets(3162,5): error MSB3821: Couldn't process file Setting.resx due to its being in the Internet or Restricted zone or
having the mark of the web on the file. Remove the mark of the web if you want to process these files. [C:\MagicRemoteS
ervice\MagicRemoteService\MagicRemoteService.csproj]
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.tar
gets(3162,5): error MSB3821: Couldn't process file TVAdder.resx due to its being in the Internet or Restricted zone or
having the mark of the web on the file. Remove the mark of the web if you want to process these files. [C:\MagicRemoteS
ervice\MagicRemoteService\MagicRemoteService.csproj]`
2. I don't have an exe in the root of the installation directory. Is that a problem or should I be using /bin/magicremoteservice.exe ?
3. I followed the instructions from your reply and still got a black screen. I'm of course not confident that I updated properly though. I couldn't see a version number; where will that be if I do it correctly?

Sorry if I'm needing very basic help here.

**Comment by Cathwyler** [comment]
I already deleted the 1.2.4.1 release version because it was a bit buggy.
Use the latest [release](https://github.com/Cathwyler/MagicRemoteService/releases) instead (sorry for that).

Apart from that, if you really want to build on your own computer, you need to unlock internet security check on each file's property and you will need to download all missing nuggets packages. But i doubt is that you really want to do.

**Comment by sethly87** [comment]
I just started fresh with the 1.2.4.3 release and it works perfectly!

For reference, I downloaded the code, unblocked the internet security on [MagicRemoteService.csproj] and a few other files giving errors when using build.bat, and it all built successfully.

For the future, what is the method to avoid downloading the packages via NuGet and building on my computer? It looked like the only option to me.
